app/src/main/java/com/bogdan/codeforceswatcher/features/problems/ProblemsFragment.kt

```diff
@@ -1,126 +1,389 @@
+        modifier = Modifier
+            .size(24.dp)
+            .clickable { onStar(problem.id) }
+    )
+}
+
+@OptIn(ExperimentalAnimationApi::class)
+@Composable
+private fun NavigationBar(
+    onFilter: () -> Unit,
+    problemsRequest: (String) -> Unit
+) {
+    val localFocusManager = LocalFocusManager.current
+    var isShownSearchTextFieldState by remember { mutableStateOf(false) }
+
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .height(56.dp)
+            .padding(horizontal = 20.dp),
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        AnimatedVisibility(
+            visible = isShownSearchTextFieldState.not(),
+            enter = fadeIn(),
+            modifier = Modifier.weight(1f)
+        ) {
+            Text(
+                text = stringResource(R.string.problems),
+                style = AlgoismeTheme.typography.headerSmallMedium,
+                color = AlgoismeTheme.colors.secondary
+            )
+        }
+
+        Row(horizontalArrangement = Arrangement.spacedBy(20.dp)) {
+            Box(
+                modifier = if (isShownSearchTextFieldState) Modifier.weight(1f) else Modifier.width(30.dp),
+                contentAlignment = Alignment.CenterEnd
+            ) {
+                this@Row.AnimatedVisibility(
+                    visible = isShownSearchTextFieldState,
+                    enter = expandIn(
+                        expandFrom = Alignment.TopEnd,
+                        initialSize = { IntSize(30, 30) },
+                        animationSpec = tween(
+                            durationMillis = 150,
+                            easing = LinearEasing
+                        )
+                    )
+                ) {
+                    SearchTextField(
+                        modifier = Modifier.fillMaxWidth(1f),
+                        keyboardOptions = KeyboardOptions(
+                            keyboardType = KeyboardType.Ascii,
+                            imeAction = ImeAction.Done
+                        ),
+                        keyboardActions = KeyboardActions(
+                            onDone = { localFocusManager.clearFocus() }
+                        ),
+                        onValueChange = { query ->
+                            problemsRequest(query)
+                        }
+                    )
+                }
+
+                Image(
+                    painter = painterResource(R.drawable.ic_search_icon),
+                    contentDescription = null,
+                    modifier = Modifier.clickable { isShownSearchTextFieldState = true }
+                )
+            }
+
+            if (isShownSearchTextFieldState) {
+                CrossButton(
+                    onCross = {
+                        isShownSearchTextFieldState = false
+                        problemsRequest("")
+                    }
+                )
+            } else {
+                FilterButton(
```

Comment:
Won't `FilterButton { onFilter() }` would be better?

Reply:
I thought about it. Now I know the best choice, thanks ;)